src/components/data-table-v2/data-table-v2.js

```diff
@@ -0,0 +1,310 @@
+    this._actionBarToggle(this.state.checkboxCount > 0);
+  };
+
+  _selectAllToggle = detail => {
+    const checked = detail.element.checked;
+
+    const inputs = [...this.element.querySelectorAll(this.options.selectorCheckbox)];
+
+    this.state.checkboxCount = checked ? inputs.length - 1 : 0;
+
+    inputs.forEach(item => {
+      item.checked = checked;
+    });
+
+    this._actionBarToggle(this.state.checkboxCount > 0);
+    this.countEl.textContent = this.state.checkboxCount;
+  };
+
+  _actionBarCancel = () => {
+    const inputs = [...this.element.querySelectorAll(this.options.selectorCheckbox)];
+
+    inputs.forEach(item => {
+      item.checked = false;
+    });
+
+    this.state.checkboxCount = 0;
+    this._actionBarToggle(false);
+    this.countEl.textContent = this.state.checkboxCount;
+  };
+
+  _actionBarToggle = toggleOn => {
+    const transition = evt => {
+      this.batchActionEl.removeEventListener('transitionend', transition);
+
+      if (evt.target.matches(this.options.selectorActions)) {
+        if (this.batchActionEl.dataset.active === 'false') {
+          this.batchActionEl.setAttribute('tabIndex', -1);
+        } else {
+          this.toolbarEl.style.visibility = 'hidden';
+          this.batchActionEl.setAttribute('tabIndex', 0);
+        }
+      }
+    };
+
+    if (toggleOn) {
+      this.batchActionEl.dataset.active = true;
+      this.batchActionEl.classList.add('bx--batch-actions--active');
+    } else {
+      this.toolbarEl.style.visibility = '';
+      this.batchActionEl.dataset.active = false;
+      this.batchActionEl.classList.remove('bx--batch-actions--active');
+    }
+
+    this.batchActionEl.addEventListener('transitionend', transition);
+  };
+
+  _expandableRowsInit = expandableRows => {
+    expandableRows.forEach(item => {
+      item.classList.remove(this.options.classExpandableRowHidden);
+      this.tableBody.removeChild(item);
+    });
+  };
+
+  _rowExpandToggle = detail => {
+    const element = detail.element;
+    const parent = eventMatches(detail.initialEvt, this.options.eventParentContainer);
+
+    const index = this.expandCells.indexOf(element);
+    if (element.dataset.previousValue === undefined || element.dataset.previousValue === 'expanded') {
+      element.dataset.previousValue = 'collapsed';
+      parent.classList.add(this.options.classExpandableRow);
+      this.tableBody.insertBefore(this.expandableRows[index], this.parentRows[index + 1]);
+    } else {
+      parent.classList.remove(this.options.classExpandableRow);
+      this.tableBody.removeChild(parent.nextElementSibling);
+      element.dataset.previousValue = 'expanded';
+    }
+  };
+
+  _expandableHoverToggle = (element, flag) => {
+    element.previousElementSibling.classList.add(this.options.classExpandableRowHover);
+
+    const mouseout = evt => {
+      element.previousElementSibling.classList.remove(this.options.classExpandableRowHover);
+    };
+
+    element.addEventListener('mouseout', mouseout);
```

Comment:
Given this line runs upon every `mouseover` event on the element, this line causes redundant `mouseout` event listeners attached to the element. Some ways to address this issues are:

* Hoisting the `mouseout` listener so that the 2nd arg of `.addEventListener()` becomes the same function ref
* Attaching the `mouseout` listener in the same way as we do for `mouseover` (Assuming that `mouseout` bubbles)
* Detaching the `mouseout` listener in the listener function itself (As you do for `focusout` below)

Reply:
Good call, I'll detach